File: Development/WoodLotFrontEnd/src/app/entita/contadino/contadino.ts
import {ProdottoOrdine} from "../prodottoOrdine/prodotto-ordine";

export class Contadino {

  public nome: String
  public cognome: String
  public listaAlberi: ProdottoOrdine[]
  public id: number
  public coordinateGeografiche: String
  public titolareConto: String
  public swift: String

  constructor(nome: String, cognome: String, listaAlberi: ProdottoOrdine[], id: number, coordinateGeografiche: String,
              titolareConto: String, swift: String) {
    this.nome = nome
    this.cognome = cognome
    this.listaAlberi = listaAlberi
    this.id = id
    this.coordinateGeografiche = coordinateGeografiche
    this.titolareConto = titolareConto
    this.swift = swift
  }
}

File: Development/WoodLotBackEnd/src/main/java/it/unisa/WoodLot/model/entity/Contadino.java
package it.unisa.WoodLot.model.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

/**
 * Classe che modella un contadino
 *
 * @author Alessia Ture
 */
@Data
@Entity
@Table(name = "contadino")
public class Contadino extends UtenteRegistrato {

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "cognome", nullable = false)
    private String cognome;

    @Column(name = "data_di_nascita", nullable = false)
    private Date dataDiNascita;

    @Column(name = "coordinate_geografiche", nullable = false)
    private String coordinateGeografiche;

    @Column(name = "swift")
    private String swift;

    @Column(name = "titolare_conto")
    private String titolareConto;

    public Contadino(){}
    public Contadino(Long id, String email, String password) {
        super(id, email, password);
    }
}
